services/SecurityService.go

```diff
@@ -0,0 +1,92 @@
+package services
+
+import (
+	"fmt"
+	"github.com/jasongauvin/wikiPattern/models"
+	"golang.org/x/crypto/bcrypt"
+	"time"
+)
+
+type RegisterForm struct {
+	Email    string `form:"userEmail" binding:"required"`
+	Password string `form:"userPassword" binding:"required"`
+}
+
+type LoginForm struct {
+	Email    string `form:"userEmail" binding:"required"`
+	Password string `form:"userPassword" binding:"required"`
+}
+
+// SaveUser creates a user.
+// Return the created object
+func SaveUser(email string, password string) (*models.User, error) {
+	var err error
+	var user models.User
+	user.Email = email
+	user.Password = password
+
+	userCreated, err := models.CreateUser(&user)
+	if err != nil {
+		return nil, err
+	}
+
+	return userCreated, nil
+}
+
+// AuthenticateUser let you find the user from its email,
+// Compare the passed password and the hashed password stored.
+// Check the session expiration date and update the token with a new one.
+// Return the UserSession.
```

Comment:
We already know the return type thanks to the function signature, redundant comment